codec-http2/src/main/java/io/netty/handler/codec/http2/Http2FlowController.java

```diff
@@ -22,6 +22,12 @@
 public interface Http2FlowController {
 
     /**
+     * Gets the {@link Http2Connection.PropertyKey} used for accessing the {@link Http2Stream.FlowControlState} property
+     * that this controller associates with each {@link Http2Stream}.
+     */
+    Http2Connection.PropertyKey stateKey();
```

Comment:
It seems like the trade off we are making is making access to the flow controller state require more work with the benefit of not having setter methods for the flow controller states on the `Http2Stream` interface?  It seems like the API became a bit less clear in the process.  Seems like to know what this is you have to understand the class hierarchy and object relationships, and timing as to when it gets set.  Doing a new implementation of the flow controller now requires you to digest this and understand they must put the state in the property map.  Perhaps I just need to hear if this is the rational you are using, or if there are other pieces I'm not considering?